routes/index.tsx

```diff
@@ -54,8 +55,17 @@ export const handler: Handlers<HomePageData, State> = {
       ctx.state.sessionId,
     );
     const lastPage = calcLastPage(allItems.length, PAGE_LENGTH);
+    const needsSetup = Deno.env.get("GITHUB_CLIENT_ID") === undefined ||
```

Comment:
Perhaps, let's move this inline with the component itself. That way, we don't have to pass it from the handler to the component. WDYT?

Reply:
Oh, I didn't think I could do that. moving into component doesn't work but I just noticed I can do that with declaring it at top level. Please review the code for any problems.